src/main/java/macrobase/analysis/MacrobaseAnalyzer.java

```diff
@@ -0,0 +1,55 @@
+package macrobase.analysis;
+
+import macrobase.analysis.outlier.OutlierClassifier;
+import macrobase.analysis.result.MacrobaseAnalysisResult;
+import macrobase.analysis.summary.Summarizer;
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+import macrobase.ingest.DataIngester;
+import macrobase.transform.FeatureTransform;
+import org.apache.commons.collections4.iterators.IteratorIterable;
+
+import java.util.Iterator;
+
+public class MacrobaseAnalyzer implements Iterator<MacrobaseAnalysisResult> {
+
+    private DataIngester ingester;
+    private FeatureTransform source;
+    private OutlierClassifier outlierClassifier;
+    private Summarizer summarizer;
+    private int windowSize;
+
+    /**
+     * MacrobaseAnalyzer will create the following attributes in the constructor:
+     *  - DataIngester ingester, that gets the data, usually just initialize with a conf, based on conf
+     *  - FeatureTransform based on conf and ingester
+     * @param conf
+     */
+    public MacrobaseAnalyzer(MacroBaseConf conf) {};
+    // Example pseudo-code,
+    // ingester = constructIngester(conf);
+    // source = new FeatureTransform(conf, ingester);
+    // windowSize = conf.getWindowSize();
+    // outlierClassifier = constructOutlierClassifier(conf);
+    // Summarizer = new Summarizer(conf, outlierClassifier);
+
+    @Override
+    public MacrobaseAnalysisResult next() {
+
+        int index = 0;
+        for (Datum nextDatum : new IteratorIterable<>(source)) {
+            summarizer.observe(nextDatum);
```

Comment:
It could. I don't like that API that much. It's too much nesting.
I'm not sure I am completely sold on the `return (new Summarizer(conf, new OutlierClassifier(conf, new BasicDataTransform(conf, new DataIngest(conf).output()).output()).output()).output()).getNext();` line.
I can't imagine where in the process would the windowing happen?
Also a method returning an Iterator seems ugly to me, but if that's a standard java thing, I'd be ok with that.